### ISSUE TITLE
Is the project still being worked on?

### ISSUE THREAD

**Issue by mitfry** [shadowmage45/ArkMapPlot#1]
Hi there ShadowMage45,

A couple of weeks back I read your post about the GUI version of the ARK tools. I absolutely loved the fact you took the initiative to work this out. I've been using your latest release for a while now and I was wondering if you still have some spare time to work on the project. I'd love to see the Center and Ragnarok maps introduced as well for example.

Kind regards,

Mitfrey

**Comment by shadowmage45** [comment]
Due to the devs' rebalancing all of the single-player stats and difficulty, and completely screwing up my save game/worlds/dino stats, I've mostly stopped playing ARK (when one of my decently-bred Rex's went from ~10khp up to 40k, it removed all difficulty, and with it my will to play).  Not sure that I'll get back into it anytime soon.

However, the base code was intended to be easily extensible.  You should only need to write up the .map file and include an image of the world map.

I'll take a look at the code to verify this, and will update once I know for certain.

**Comment by shadowmage45** [comment]
Indeed;  the code loads map data for all all *.map files that it locates.  -- https://github.com/shadowmage45/ArkMapPlot/blob/master/ArkMapPlot/MainWindow.xaml.cs#L63-L68

All one would need to do in order to add support for Center and Ragnarok would be to create the *.map definition file, add the image of the map, and set up all of the directories properly.  Examples of the layout/setup can be seen in the existing Island and ScorchedEarth data folders.

**Comment by mitfry** [comment]
Thanks for the blazing fast reply!

I understand. Definitely not a fan of the "re-balancing" attempts they've done so far either. Maybe you'll find some challenge in the latest "end-game" content like the TEK cave.

I do have some coding experience so if it turns out to be easily extensible and you don't want to spend the time doing it yourself, which I completely understand if you're not playing ARK, I might be able to do  it myself. 

I would like to thank you in advance for your time!

**Comment by shadowmage45** [comment]
For adding the new maps -- you shouldn't even need to do any coding.  The existing .exe should work fine as soon as the data / config files for the new maps are added :)

Yeah, I'll probably give Ark another try sometime over the winter months now that it is in full release.

If you run into questions/problems during your attempts to add the new maps, feel free to open up an issue ticket on it, or even reply to this one.  I get instant email updates for issue tickets on any of my repositories, so should be a good method to get ahold of me if needed.

**Comment by mitfry** [comment]
Perfect! When I have some spare time and get some additions working OR need a little help doing so I'll definitely be in touch, thank you.

Talk to you later :)

**Comment by mitfry** [comment]
Hello Shadowmage!

Today I got off work a lot earlier than expected so I spend some time on extending the tool's functions :). I was pleasantly surprised by how easy it was to add the maps, just like you said it'd be! 

I managed to add the following:
* Added Ragnarok map support
* Added theCenter map support
* Added the Otter creature.
* Added the Phoenix creature.
* Fixed Ptera and Moth being data being mixed up because of the data in ark_data_en.json

I would love to have a "refresh" button in the tools GUI so I don't have to re-open the tool every time new data is generated. Could even work like this:
1. Have it check for new data every X amount of minutes/seconds. (Refresh button grey).
2. Refresh button becomes clickable when newer JSON files are found in the map's data folder.
3. A checkbox next to the Refresh button to active an "Auto Refresh" which activates automatically after step 2.

OR

Have the tool run a batch file in the background that refreshes the data for the chosen map when the refresh button is clicked, of course followed by a reload of the data.

What do you think? :)

**Comment by shadowmage45** [comment]
Glad to hear; I try and make most code that I write configurable, and extensible through config files where possible.

Refresh button (reload data from current folder) -- should be easy enough to add that in.  Will look into this for a near-future update, I would guess within the next week or so.

Auto-Refresh -- possibly.  The hardest part would be properly detecting file changes, so things are only reloaded when necessary.  Could perhaps implement a quick-and-dumb timer based auto-refresh capability that simply reloads the data every X minutes/seconds (configurable).  If/when I am working on the refresh button, I'll know more about what this would take to set up.

Update button (interface with ark-tools to export new data + reload the new data) -- doable, but would require a bit more effort to setup the interaction between the two programs.  Will certainly investigate this if I get back into Ark much.  For now I think creating a batch file to update the .json combined with the refresh button will be workable.

**Comment by mitfry** [comment]
Thanks for the reply!

Sounds good, as long as it doesn't take too much of your time :).

I'm currently using a self-made batch file that asks for the server and whether I want to export tamed or wild dino's from the server' save file I chose. I've pointed it to the dinoData folder so it'll automatically refresh all the files in there. When it's finished I open up the tool and read the data I want to see.

Since all files in the dinoData folder are updated with these commands, I'd say that all the tool would have to do is get the "created" time-stamp of one of the files in the dinoData folder at startup and save it somewhere to be used in a comparison function later. This value can later be overwritten when a new "latest" time-stamp is retrieved. Not sure how hard this is to do though.

Could you explain how these two programs would need to interact? I'm not as experienced as you in coding so maybe I underestimated it. I thought of it like this;

1. (bad) We create different batch files and have the tool simply execute them upon clicking some button. The batch files however won't work for different users, without them editing the batch files to match with their folder structure.

2. (better)We allow certain paths to be set in the tool and have the tool create a custom batch file using this info, run the batch file and then delete it after it was executed.

Is this something that's even worth giving a try and could be done in an amount of time that is in your case reasonable to spend on it, while you're not into ARK? Otherwise I might have to try myself, but it's been years since I've done any C coding so it would take a while for me to get back into it. Still, it'd be a fun challenge!

**Comment by shadowmage45** [comment]
Refresh button should be in place -- Edit -> Refresh Map.  ( https://github.com/shadowmage45/ArkMapPlot/commit/372762ef1780cd6a93babcca91557b4be7565dc3 )

**Comment by mitfry** [comment]
Awesome ShadowMage! Thank you :) Will take a look tomorrow!

**Comment by mitfry** [comment]
Alright ShadowMage, it got a little later than "tomorrow" as I am incredibly busy irl. I am excited but have to hold myself back hehe. Priorities.. right ;)

I downloaded and compiled the project. The reload button seems to work fine. Meanwhile I discovered selecting the same map from File -> Load Map -> "Currently already selected map" also performs a reload. Is this intended?

Thanks again for the addition :)
